scraper/github/app/utils/github_requests.py

```diff
@@ -3,132 +3,191 @@
 from app.settings import GITHUB_API, GITHUB_TOKEN
+from datetime import date
 
+logging.basicConfig(level=logging.DEBUG)
 
-def request_failed(ret):
-    return ret and "status" in ret and ret["status"] == "error"
-
-def status_check(r):
-    """
-
-    Just return a message when the request's status is not 200
 
-    @params : the request object
-    @returns : a tuple (boolean, object),
-                the boolean define if the request gone well and
-                the object is detail of error
+class GithubClient:
     """
-
-    if r.status_code != 200:
-        return (
-            False,
-            {
-                "status": "error",
-                "url": r.url,
-                "payload": json.loads(r.content),
-                "message": "Something went wrong with the request",
-                "code": r.status_code,
-            },
-        )
-    else:
-        return True, {}
-
-
-def get_users(
-    pagination_limit: int = 2, on_pageloaded_success=None, on_pageloaded_error=None
-):
+    This is a client to access Github API
     """
 
-    This method will just fetch users from
-    github api that are in location cameroon or cameroun
-
-    @params : pagination_limit [the number of page we want to fetch]
-    @returns : users [json array of users]
-
-    """
-
-    users = []
-
-    for page in range(1, pagination_limit):
-        # We set our query
-        query = "location:%22cameroon%22+location:%22cameroun%22&page={}".format(page)
-
-        # a simple request to the api
-        headers = {"Authorization": "token {}".format(GITHUB_TOKEN)}
-        r = requests.get("{}/search/users?q={}".format(GITHUB_API, query), headers=headers)
-        # We check the status of the requet and return a predefined error message
-        schk = status_check(r)
+    def __init__(self):
+        self.GITHUB_API = GITHUB_API
+        self.GITHUB_TOKEN = GITHUB_TOKEN
+        self.session: requests.Session = requests.Session()
+        self.session.headers.update({"Authorization": f"token {self.GITHUB_TOKEN}"})
+        self.logger = logging.getLogger("")
+
+        # configure a console handler
+        # console = logging.StreamHandler()
+        # console.setLevel(logging.INFO)
+        # self.logger.addHandler(console)
+
+    def request_failed(self, ret: Dict) -> bool:
+        return ret and "status" in ret and ret["status"] == "error"
+
+    def status_check(self, r: requests.Response) -> Tuple[bool, Dict]:
+        """
+
+        Just return a message when the request's status is not 200
+
+        @params : the request object
+        @returns : a tuple (boolean, object),
+                    the boolean define if the request gone well and
+                    the object is detail of error
+        """
+
+        if r.status_code != 200:
+            return (
+                False,
+                {
+                    "status": "error",
+                    "url": r.url,
+                    "payload": json.loads(r.content),
```

Comment:
can we use this `r.json()["items"]` here as well ?

Reply:
i think not because "items" is not a valid key in case of an error